test/helpers/mcd_config_drift.go

```diff
@@ -0,0 +1,299 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	configDriftFilename             string = "/etc/etc-file"
+	configDriftExpectedFileContents string = "expected-file-data"
+	nodeAnnotationKey               string = "mcdtest"
+)
+
+// RunConfigDriftTest does the following:
+// 1. Creates a MachineConfig whose application is deferred to the setup
+// function. This is because the setup function may need to set up other things
+// such as the MachineConfigPool, label the node, etc. This allows reuse in
+// both a highly available context as well as single-node context.
+// 2. Mutates the file specified by the MachineConfig on the target node so its
+// contents do not match what the MachineConfig specifies.
+// 3. Verifies that the target node and MachineConfigPool become degraded.
+// 4. Mutates the file specified by the MachineConfig to make its contents
+// equal to that specified by the MachineConfig.
+// 5. Verifies that the target node and MachineConfigPool recover from this.
+// 6. Again mutates the file specified by the MachineConfig on the target node so its
+// contents do not match what the MachineConfig specifies.
+// 7. Sets the forcefile on the target node.
+// 8. Verifies that the target node and MachineConfigPool recover from this.
+//
+// The caller must provide the setup / teardown functions since it has the most
+// context for what is required.
+func RunConfigDriftTest(t *testing.T, mcpName string, setup func(*mcfgv1.MachineConfig), teardown func()) {
+	cs := framework.NewClientSet("")
+
+	// Create an MCO-owned file that we'll mutate later
+	mcadd := CreateMCToAddFileForRole("mcd-config-drift", mcpName, configDriftFilename, configDriftExpectedFileContents)
+
+	// Add kernel arguments onto our MachineConfig so we can verify they're
+	// present pre-and-post recovery
+	mcadd.Spec.KernelArguments = []string{"foo=bar", "foo=baz"}
+
+	// Run the provided setup function to apply the MachineConfig and do any other required setup.
+	setup(mcadd)
+	defer teardown()
+
+	// Get the MCD pod that is running on that node.
+	// Note: This could have changed between when we applied our MC and now since
+	// the node may have rebooted.
+	targetNode := GetSingleNodeByRole(t, cs, mcpName)
+
+	// Ensure our node annotation is removed at the end of the test
+	defer mutateNode(t, cs, targetNode, func(n *corev1.Node) {
+		delete(n.Annotations, nodeAnnotationKey)
+	})
+
+	// Get the target MachineConfigPool
+	targetMCP, err := cs.MachineConfigPools().Get(context.TODO(), mcpName, metav1.GetOptions{})
+	require.Nil(t, err)
+
+	// Get the MCD pod for our target node
+	mcdPod, err := MCDForNode(cs, &targetNode)
+	require.Nil(t, err)
+
+	// Check that we have our kernel args pre-recovery
+	assertHasKernelArgs(t, cs, targetNode, mcadd.Spec.KernelArguments)
+
+	// Get our kernel args for future verification
+	kargs := getKernelArgs(t, cs, targetNode)
+
+	t.Logf("Target MCD Pod: %s. To stream pod logs, copy / paste the below:", mcdPod.Name)
```

Comment:
```suggestion
	t.Logf("Target MCD Pod: %s", mcdPod.Name)
```

Reply:
Beyond my current development needs, we don't. I'll take this out.